src/Classes/Canvas.ts

```diff
@@ -154,7 +158,159 @@ export class Canvas {
             this.paint();
         } else {
             if (this.context) {
-                this.context.clearRect(0, 0, this.dimension.width, this.dimension.height);
+                CanvasUtils.clear(this.context, this.dimension);
+            }
+        }
+    }
+
+    public drawPolygonMask(rawData: ICoordinates[]): void {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of public instance method not allowed after declaration of private instance method. Instead, this should come after constructors.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5171185)